app/src/main/java/com/github/se/cyrcle/CyrcleNavHost.kt

```diff
@@ -0,0 +1,39 @@
+import androidx.compose.runtime.Composable
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.navigation
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.ui.add.AttributesPicker
+import com.github.se.cyrcle.ui.add.LocationPicker
+import com.github.se.cyrcle.ui.authentication.SignInScreen
+import com.github.se.cyrcle.ui.card.CardScreen
+import com.github.se.cyrcle.ui.list.SpotListScreen
+import com.github.se.cyrcle.ui.map.MapScreen
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.navigation.Route
+import com.github.se.cyrcle.ui.navigation.Screen
+
+@Composable
+fun CyrcleNavHost(
+    navigationActions: NavigationActions,
+    navController: NavHostController,
+    parkingViewModel: ParkingViewModel
+) {
+  NavHost(navController = navController, startDestination = Route.AUTH) {
+    navigation(startDestination = Screen.AUTH, route = Route.AUTH) {
+      composable(Screen.AUTH) { SignInScreen(navigationActions) }
+    }
+    navigation(startDestination = Screen.LIST, route = Route.LIST) {
+      composable(Screen.LIST) { SpotListScreen(navigationActions, parkingViewModel) }
+      composable(Screen.CARD) { CardScreen(navigationActions, parkingViewModel) }
+    }
+    navigation(startDestination = Screen.MAP, route = Route.MAP) {
+      composable(Screen.MAP) { MapScreen(navigationActions) }
+    }
+    navigation(startDestination = Screen.LOCATION_PICKER, route = Route.ADD_SPOTS) {
```

Comment:
I think that each route should start from a TLD , here it would `Map` . This ensures that the navigation flow stay logical and consistent as a graph. Now , it is a comment about semantic and it functionally works so i won't request changes.

Reply:
To be discussed with the team, I don't think that is how Route works, otherwise all our screen that have nothing in common appart being accessible from the MapScreen would be in the same group ?